src/utils.ts

```diff
@@ -192,22 +190,50 @@ export async function fromPublicDirectory(
   modelPath: string,
   publicPath: string,
   forceWrite: boolean,
-  version: number
+  version: number,
+  numFetchReties: number
 ) {
   const pvFile = await open(modelPath, 'w');
   if (
     forceWrite ||
     pvFile.meta === undefined ||
     version > pvFile.meta.version
   ) {
-    const response = await fetch(publicPath, {
-      cache: 'no-cache',
-    });
-    if (!response.ok) {
-      throw new Error(`Failed to get model from '${publicPath}'`);
+    if (numFetchReties < 0) {
+      throw Error('numFetchRetries must be a positive number');
+    }
+
+    let numAttemptsLeft: number = numFetchReties + 1;
+    let data: ArrayBuffer = null;
+    let error: Error = null;
+    while (data === null && numAttemptsLeft > 0) {
+      error = null;
+      try {
+        const response = await fetch(publicPath, {
+          cache: 'no-cache',
+        });
+        if (response.ok) {
+          data = await response.arrayBuffer();
+        } else {
+          const responseText = await response.text();
+          error = new Error(
+            `Error response returned while fetching model from '${publicPath}': ${responseText}`
+          );
+        }
+      } catch (e: any) {
+        error = new Error(
+          `Failed to fetch model from '${publicPath}': ${e.message}`
+        );
+      }
+
+      numAttemptsLeft--;
+    }
+
+    if (data !== null) {
+      await pvFile.write(new Uint8Array(data), version);
+    } else if (error !== null) {
```

Comment:
is it possible to have no data `( data !== null)` and still doesn't return any error since `error !== null`?

Reply:
I don't think that's possible because `response.arrayBuffer()` can't return a `null`, but I'll add an `else` to this block that throws a generic error just in case.